modules/apps/layout/layout-content-page-editor-web-test/src/testIntegration/java/com/liferay/layout/content/page/editor/web/internal/util/SegmentsExperienceTestUtil.java

```diff
@@ -0,0 +1,119 @@
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.segments.constants.SegmentsEntryConstants;
+
+import java.io.InputStream;
+
+import java.util.List;
+
+import org.junit.Assert;
+
+/**
+ * @author Mikel Lorza
+ */
+public class SegmentsExperienceTestUtil {
+
+	public static void addSegmentsExperienceData(
+			String fileName, Layout layout,
+			LayoutPageTemplateStructureLocalService
+				layoutPageTemplateStructureLocalService,
+			LayoutPageTemplatesImporter layoutPageTemplatesImporter,
+			long segmentsExperienceId)
+		throws Exception {
+
+		LayoutPageTemplateStructure layoutPageTemplateStructure =
+			layoutPageTemplateStructureLocalService.
+				fetchLayoutPageTemplateStructure(
+					layout.getGroupId(), layout.getPlid());
+
+		String segmentsExperienceData =
+			layoutPageTemplateStructure.getDefaultSegmentsExperienceData();
+
+		LayoutStructure layoutStructure = LayoutStructure.of(
+			segmentsExperienceData);
+
+		String pageElementJSON = _readFileContent(fileName);
+
+		if (segmentsExperienceId != SegmentsEntryConstants.ID_DEFAULT) {
+			layoutPageTemplatesImporter.importPageElement(
+				layout, layoutStructure, layoutStructure.getMainItemId(),
+				pageElementJSON, 0, segmentsExperienceId);
+		}
+		else {
+			layoutPageTemplatesImporter.importPageElement(
+				layout, layoutStructure, layoutStructure.getMainItemId(),
+				pageElementJSON, 0);
+		}
+	}
+
+	public static void checkSegmentsExperiences(
+		Layout layout, long sourceSegmentsExperienceId,
+		long targetSegmentsExperienceId) {
+
+		List<FragmentEntryLink> sourceFragmentEntryLinks =
+			FragmentEntryLinkLocalServiceUtil.
+				getFragmentEntryLinksBySegmentsExperienceId(
+					layout.getGroupId(), sourceSegmentsExperienceId,
+					layout.getPlid());
+
+		List<FragmentEntryLink> targetFragmentEntryLinks =
+			FragmentEntryLinkLocalServiceUtil.
+				getFragmentEntryLinksBySegmentsExperienceId(
+					layout.getGroupId(), targetSegmentsExperienceId,
+					layout.getPlid());
+
+		Assert.assertTrue(sourceFragmentEntryLinks.size() == 1);
+		Assert.assertTrue(targetFragmentEntryLinks.size() == 1);
+
+		FragmentEntryLink sourceFragmentEntryLink =
+			sourceFragmentEntryLinks.get(0);
+
+		FragmentEntryLink targetFragmentEntryLink =
+			targetFragmentEntryLinks.get(0);
+
+		Assert.assertEquals(
+			sourceFragmentEntryLink.getCss(), targetFragmentEntryLink.getCss());
+		Assert.assertEquals(
+			sourceFragmentEntryLink.getHtml(),
+			targetFragmentEntryLink.getHtml());
+		Assert.assertEquals(
+			sourceFragmentEntryLink.getJs(), targetFragmentEntryLink.getJs());
+		Assert.assertEquals(
+			sourceFragmentEntryLink.getConfiguration(),
+			targetFragmentEntryLink.getConfiguration());
+		Assert.assertEquals(
+			sourceFragmentEntryLink.getEditableValues(),
+			targetFragmentEntryLink.getEditableValues());
+	}
+
+	private static String _readFileContent(String fileName) throws Exception {
```

Comment:
I think that we don't need a method, we can use:

StringUtil.read(getClass(), "dependencies/timer-tasks-workflow-definition.xml");

Reply:
Changed ❤️